rs/sns/swap/src/swap.rs

```diff
@@ -894,9 +894,24 @@ impl Swap {
         // for all NF investors.
         let mut global_cf_memo: u64 = NEURON_BASKET_MEMO_RANGE_START;
         for cf_participant in self.cf_participants.iter_mut() {
+            let controller = cf_participant.try_get_controller();
+
             for cf_neuron in cf_participant.cf_neurons.iter_mut() {
                 // Create a closure to ensure `global_cf_memo` is incremented in all cases
                 let mut process_cf_neuron = || {
+                    let controller = match controller.clone() {
```

Comment:
Isn't controller of type PrincipalId? And isn't that Copy? Shouldn't clippy say that clone is unecesary here?

Reply:
the `controller` being matched upon is type `Result<PrincipalId, String>` which is not `Copy`